packages/protocol/contracts/tokenvault/BridgedERC20.sol

```diff
@@ -65,13 +73,25 @@ contract BridgedERC20 is BridgedERC20Base, IERC20MetadataUpgradeable, ERC20Upgra
         // Initialize OwnerUUPSUpgradable and ERC20Upgradeable
         __Essential_init(_addressManager);
         __ERC20_init({ name_: _name, symbol_: _symbol });
+        __ERC20Snapshot_init();
+        __ERC20Votes_init();
 
         // Set contract properties
         srcToken = _srcToken;
         srcChainId = _srcChainId;
         srcDecimals = _decimals;
     }
 
+    /// @notice Creates a new token snapshot.
+    function snapshot() public onlyOwner {
+        _snapshot();
+    }
+
+    /// @notice Delegate votes from the original sender (tx.origin) to `delegatee`.
+    function delegateByTxOrigin(address delegatee) public virtual {
+        _delegate(tx.origin, delegatee);
```

Comment:
I don't think it's a good idea to use tx.origin here. Maybe there is another way.